crates/miden-tx/src/executor/mod.rs

```diff
@@ -309,7 +313,10 @@ where
         notes: InputNotes<InputNote>,
         tx_args: TransactionArgs,
         source_manager: Arc<dyn SourceManager>,
-    ) -> Result<NoteAccountExecution, TransactionExecutorError> {
+    ) -> Result<NoteConsumptionExecution, TransactionExecutorError> {
+        let mut notes_map: BTreeMap<_, _> =
+            notes.iter().map(|note| (note.id(), note.clone().into_note())).collect();
```

Comment:
@bobbinth I have done this to avoid modifying `TransactionExecutorHost` to use `Note` instead of `NoteId` which I presume is not possible given it starts to involve VM execution AFAICT.